packages/react-devtools-scheduling-profiler/src/view-base/HorizontalPanAndZoomView.js

```diff
@@ -30,97 +31,84 @@ import {
   zoomState,
 } from './utils/scrollState';
 import {
-  DEFAULT_ZOOM_LEVEL,
   MAX_ZOOM_LEVEL,
   MIN_ZOOM_LEVEL,
   MOVE_WHEEL_DELTA_THRESHOLD,
 } from './constants';
 
-export type HorizontalPanAndZoomViewOnChangeCallback = (
-  state: ScrollState,
-  view: HorizontalPanAndZoomView,
-) => void;
-
 export class HorizontalPanAndZoomView extends View {
+  _contentView: View;
   _intrinsicContentWidth: number;
   _isPanning = false;
-  _scrollState: ScrollState = {offset: 0, length: 0};
-  _onStateChange: HorizontalPanAndZoomViewOnChangeCallback = () => {};
+  _viewState: ViewState;
 
   constructor(
     surface: Surface,
     frame: Rect,
     contentView: View,
     intrinsicContentWidth: number,
-    onStateChange: HorizontalPanAndZoomViewOnChangeCallback,
+    viewState: ViewState,
   ) {
     super(surface, frame);
-    this.addSubview(contentView);
-    this._intrinsicContentWidth = intrinsicContentWidth;
-    this._setScrollState({
-      offset: 0,
-      length: intrinsicContentWidth * DEFAULT_ZOOM_LEVEL,
-    });
-
-    this._onStateChange = onStateChange;
-  }
 
-  setFrame(newFrame: Rect) {
-    super.setFrame(newFrame);
+    this._contentView = contentView;
+    this._intrinsicContentWidth = intrinsicContentWidth;
+    this._viewState = viewState;
 
-    // Revalidate scrollState
-    this._setStateAndInformCallbacksIfChanged(this._scrollState);
-  }
+    viewState.onHorizontalScrollStateChange(scrollState => {
+      this.zoomToRange(scrollState.offset, scrollState.length);
+    });
 
-  setScrollState(proposedState: ScrollState) {
-    this._setScrollState(proposedState);
+    this.addSubview(contentView);
   }
 
   /**
-   * Just sets scroll state. Use `_setStateAndInformCallbacksIfChanged` if this
-   * view's callbacks should also be called.
+   * Just sets scroll state.
+   * Use `_setStateAndInformCallbacksIfChanged` if this view's callbacks should also be called.
    *
    * @returns Whether state was changed
    * @private
```

Comment:
idk if this comment has anything meaningful effect, but is it still meant to be @private?

Reply:
This was just a formatting change 😁

Reply:
oh i guess my comment got weirdly positioned, but I guess I mean since `_setScrollState` is now called `setScrollState` if it's meant to remain as `@private`. also looks like the @return value is also outdated now? in any case, seems better to rely on Flow

Reply:
Yeah good call. I didn't even notice there were Java-doc-style comment pragmas. I'm used to just ignoring those I guess in favor of the Flow types.